document-store/src/integrationTest/java/org/hypertrace/core/documentstore/mongo/MongoQueryExecutorIntegrationTest.java

```diff
@@ -278,6 +277,26 @@ public void testAggregateWithNestedFields() throws IOException {
     assertDocsEqual(resultDocs, "mongo/aggregate_on_nested_fields_response.json");
   }
 
+  @Test
+  public void testDistinctCount() throws IOException {
+    Query query =
+        Query.builder()
+            .addSelection(
+                AggregateExpression.of(DISTINCT_COUNT, IdentifierExpression.of("quantity")),
+                "qty_count")
+            .addSelection(IdentifierExpression.of("item"))
+            .addAggregation(IdentifierExpression.of("item"))
```

Comment:
Can we also test with multiple group by columns and without an alias?

Reply:
Will do.